### PR TITLE
test: add integration test for incompatible command detection with backups enabled

### PR DESCRIPTION
**Pull request by stevenpyzhang**
### Description 
There's a bug in 0.14.0 that causes the server to not enter degraded state when backups are enabled.
The bug is fixed in https://github.com/confluentinc/ksql/pull/6524 (removing deserialization from the CommandTopicBackupImpl)

This adds an integration test to check that incompatible commands are properly processed.
Also, enhanced the corruption integration test and cleaned up some other tests.

### Testing done 
test only change

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestIntegrationTestUtil.java [hunk 1]
```diff
@@ -71,6 +72,19 @@
   private RestIntegrationTestUtil() {
   }
 
+  public static HealthCheckResponse makeHealthCheck(final TestKsqlRestApp restApp) {
```
Comment (spena) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestIntegrationTestUtil.java:75]: For one moment, I got confused with the name thinking you were creating a health check object.

```suggestion
  public static HealthCheckResponse checkServerHealth(final TestKsqlRestApp restApp) {
```

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/BackupRollbackIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        ProducerConfig.ACKS_CONFIG,
+        "all"
+    );
+    final KafkaProducer<CommandId, Command> producer = new KafkaProducer<>(
+        kafkaProducerProperties,
+        InternalTopicSerdes.serializer(),
+        InternalTopicSerdes.serializer()
+    );
+
+    produceToCommandTopic(producer, commandId, command);
```
Comment (spena) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/BackupRollbackIntegrationTest.java:157]: Can all this be wrapped inside the `produceToCommandTopic`?

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestIntegrationTestUtil.java (hunk 1 above)
Comment (stevenpyzhang) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestIntegrationTestUtil.java:75]: done

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/BackupRollbackIntegrationTest.java (hunk 2 above)
Comment (stevenpyzhang) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/BackupRollbackIntegrationTest.java]: done

**Review by spena — APPROVED** [review]
lgtm

**Pull request merged**